http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/package.scala

```diff
@@ -93,6 +94,34 @@ package object http {
       Sync[F].delay(span.end())
     )
 
+  /**
+   * Creates a child Span from the parent span in the AppContext. If no parent span exists, a root
+   * Span is created. Returns a Resource which closes the Span.
+   *
+   * Usage:
+   *
+   * def myMethod(implicit ev: Ask[F, AppContext]): F[Unit] = {
+   *   // The provided AppContext has a parent Span
+   *   childSpan("myChildMethod").use { implicit ev: Ask[F, AppContext] =>
+   *     // The AppContext in this scope includes a child Span named "myChildMethod"
+   *     myChildMethod() // Returns F[Unit]
+   *   }
+   * }
+   */
+  def childSpan[F[_]: Sync](name: String)(implicit ev: Ask[F, AppContext]): Resource[F, Ask[F, AppContext]] =
+    for {
+      ctx <- Resource.eval(ev.ask)
+      newSpan <- Resource.make[F, Span](
+        Sync[F].delay(
+          Tracing.getTracer
+            .spanBuilderWithExplicitParent(name, ctx.span.orNull)
+            .setSampler(Samplers.alwaysSample())
```

Comment:
Do we think `alwaysSample` is safe? Leo pubsub messages are not terribly high

Reply:
can you just omit this?

Reply:
Done - i see the frequency is actually configurable in terra-helmfile